docs/source/_src_user_guide/state_propagation/propagation_setup/propagation_architecture.rst

```diff
@@ -0,0 +1,81 @@
+
+* Check that the independent variable or state are neither infinity nor NaN. If either one is, an exception is thrown.
+* The current state of the environment from the previous time step (translational state; orientation, altitude, etc. of bodies) are cleared, ensuring that they are recomputed for the current time step.  (see :ref:`environment_during_propagation` for the properties of bodies that are set during a time step and how to retrieve them).
+* The propagated state vector :math:`\mathbf{x}` is split into its constituent parts (*e.g.* translational, rotational, *etc.* states of separate bodies that are propagated).
+* The separate propagated states, in 'propagated coordinates', are converted to 'conventional coordinates' (see :ref:`conventional_propagated_states`). For instance: if Kepler elements of the Moon w.r.t.t the Earth are propagated, this step converts those Kepler elements (propagated state) to Cartesian elements (conventional state)
+* Each propagated state, in conventional coordinates, is set as the current state of the :class:`~tudatpy.numerical_simulation.environment.Body` object. Consequently, when the :attr:`~tudatpy.numerical_simulation.environment.Body.state` attribute of a propagated body is called during the propagation (see :ref:`environment_during_propagation`) the state from the state vector, converted to Cartesian elements (if needed) is retrieved. If the propagation origin is different from the global frame origin (see :ref:`translational_reference_frames`), a frame translation is applied before updating the body's current state. For instance, if the Kepler elements of the Moon w.r.t. Earth are propagated (propagation origin: Earth), and the global frame origin is the SSB, the :attr:`~tudatpy.numerical_simulation.environment.Body.state` attribute of the Moon will be the Cartesian state w.r.t. the SSB.
+* The time-dependent properties of the environment are updated to the current time and propagated state. Only those time/state-dependent models that are needed for either the dynamics or the dependent variables are updated during each time step. For instance, if Jupiter is a body in the environment, but Jupiter's state plays no role in either the dynamics or in the dependent variables that are saved, its state is *not* updated at each time step.
+* Each state derivative model (acceleration, torque, etc.) required for the calculation of the state derivative is evaluated. If variational equations are required, the state derivative partials are evaluated
+* The derivative of each propagated state :math:`\mathbf{x}_{i}` is evaluated from the separate state derivatives (e.g. accelerations are used to compute derivative of Kepler elements, if propagating Kepler elements), and concatenated into the complete state derivative vector :math:`\dot{\mathbf{x}}`
+
```

Comment:
This might require a bit more work (so maybe for future iterations of the documentation), but it could be nice to illustrate this section (and the single time step one) with some kind of flowchart, for instance using the "propagating keplerian elements of the Moon" example mentioned throughout this paragraph(?)